Weekly Refactoring/resolve all eslint issues and remove inline disables

bonomoon opened the pull request:
I will send a weekly PR for refactoring work on every meeting.

Please check and review them.

Thank you.

bonomoon committed:
Remove this conditional block that unnecessarily checks for window being a DIView

bonomoon committed:
Remove eslint-disable-next-line node/prefer-global/console error

bonomoon committed:
chore: solve @ts-ignore(window.body does not exist) and remove deprecated Keyboard event code
- fix window.body to document.body
- use KeyboardEvent.key(String type)

bonomoon committed:
chore: remove eslint class-method-use-this comment on closeWindow in Desk.ts, because it is already used

bonomoon committed:
chore: remove eslint class-method-use-this comment on bringWindowFront method in Desk.ts, because it is already used and @ts-ignore
- TODO: What is the type of window in argument of this method?
- TODO: Do we mean "delete" instead of "deleted"?

bonomoon committed:
chore: add a type to Desk class attributes

bonomoon committed:
chore: remove @typescript-eslint/no-unsafe-argument

bonomoon committed:
chore: use explicit type instead of any

bonomoon committed:
chore: remove eslint-disable-next-line @typescript-eslint/no-unsafe-call and convert to es6 syntax in startDarg method, Desk.ts
- no reason to comment for eslint-disable-next-line

bonomoon committed:
chore: remove eslint-disable-next-line @typescript-eslint/no-unsafe-call in startDarg method, Desk.ts
- app.dragOn: no reason to comment for eslint-disable-next-line
- app.dragLeft: add a method signature to Application.ts

bonomoon committed:
chore: add DeskEventListener type in DeskEvent.ts and change eventListener attribute according to MDN docs

bonomoon committed:
Merge remote-tracking branch 'origin/develop' into refactoring/resolve-all-eslint-issues-and-remove-inline-disables

# Conflicts:
#	System/Desk/Desk.ts
#	System/Secretary/Application.ts
#	System/Secretary/Secretary.ts

bonomoon committed:
fix: use XMLHttpRequest to get status and response object in Secretary.ts

bonomoon committed:
chore: clean up in DIWorkSpaceDock.ts
- update using es6 syntax
- remove eslint-disable-next-line @typescript-eslint/no-loop-func

bonomoon committed:
Merge branch 'develop' into refactoring/resolve-all-eslint-issues-and-remove-inline-disables

mattddean commented on System/Desk/DIWorkSpaceDock.ts:
```diff
@@ -118,27 +118,19 @@ export class DIWorkSpaceDock extends DIView {
-                        secretaryInstance.workSpaces[i].desk.workSpaceDock.clicked(this);
-                    })
-                );
+        secretaryInstance.workSpaces.forEach((_workSpace: WorkSpace, idx: number) => {
```
Unnecessary type annotations

`secretaryInstance.workSpaces.forEach((_workSpace, idx) => { ` should work

mattddean replied:
Only begin a variable with `_` when it's not used

mattddean commented on System/Desk/Desk.ts:
```diff
@@ -207,29 +204,22 @@ export class Desk {
-                secretaryInstance.mainWorkSpace.apps[i].dragStart(clipboard);
-            }
-        }
+        secretaryInstance.mainWorkSpace.apps.forEach((app) => app.allowDrag && app.dragStart(clipboard));
```
I prefer `for/const` loops over `.forEach` loops

```ts
for (const app of secretaryInstance.mainWorkSpace.apps) {
  if (app.dragStart) app.dragStart(clipboard);
}
```

What are your thoughts @qlqhwb?

qlqhwb replied:
I do not have preferences, but we can talk about convention in the meeting

qlqhwb replied:
@bonomoon I understand why you wrote this way. But I think it can be a bit misleading.
So, I think it would be better to leave it with if statement.

mattddean commented on System/Desk/DIWorkSpaceDock.ts:
```diff
@@ -118,27 +118,19 @@ export class DIWorkSpaceDock extends DIView {
-                        secretaryInstance.workSpaces[i].desk.workSpaceDock.clicked(this);
-                    })
-                );
+        secretaryInstance.workSpaces.forEach((_workSpace: WorkSpace, idx: number) => {
```
I think we can replace this `.forEach` loop with. I know that I said we could use `.forEach` loops, but I forgot about this `.entries` trick.

`for (const [workSpace, idx] of secretaryInstance.workSpaces.entries())`

What are your thoughts @qlqhwb?

qlqhwb replied:
same as above

mattddean commented on System/Desk/Desk.ts:
```diff
@@ -243,11 +233,10 @@ export class Desk {
             document.body,
             "mousemove",
-            (evt: { clientX: number; clientY: number }) => {
+            (evt: MouseEvent) => {
```
The `MouseEvent` type annotation shouldn't be necessary

qlqhwb replied:
actually, I prefer to explicitly say the type all the time, but we can talk about the convention in the meetings

mattddean commented on System/Desk/Desk.ts:
```diff
@@ -280,13 +267,12 @@ export class Desk {
             document.body,
             "mouseup",
-            (evt: { clientX: number; clientY: number }) => {
+            (evt: MouseEvent) => {
```
Unnecessary `MouseEvent` type annotation

qlqhwb replied:
same as above

mattddean reviewed (approved):
This looks really great!

bonomoon committed:
chore:resolve eslint-disable-next-line @typescript-eslint/no-unsafe-call and no-unsage-argument problem in DeskMenu
- TODO: What is function listView argument type? not DILIstView?

bonomoon committed:
chore: resolve eslint-disable-next-line @typescript-eslint/naming-convention,class-methods-use-this in DeskMenu
- just not used method

bonomoon committed:
chore: resolve eslint-disable-next-line @typescript-eslint/no-unsafe-call in Desk
- Add dragLeft abstract function
- Convert to arrow function in setTimeout
- Convert to Array.foreach

bonomoon committed:
chore: resolve eslint-disable-next-line in DIButton
- specify return type of width getter

bonomoon committed:
chore: resolve eslint-disable-next-line class-methods-use-this in DIWindow
- throw error if you invoke didMoveToParent, because this method is not implemented

bonomoon committed:
chore: remove unnecessary eslint-disable comments in DIWindow

bonomoon committed:
chore: resolve class-methods-use-this(eslint-disable) in DIView
- throw error if you use didMoveToParent

bonomoon committed:
chore: resolve class-methods-use-this, it is not necessary in DIView

bonomoon committed:
fix: resolve object as class attribute reallocation problem(ts-ignore) in DIView

bonomoon committed:
chore: use todo annotation for jsdocs

bonomoon committed:
chore: clean up constructor method in DIListView

bonomoon committed:
fix: solve ts-ignore problem according to handling user-select css option in js, DIListView

bonomoon committed:
chore: resolve @typescript-eslint/no-unsafe-call in DIDropDownView

bonomoon committed:
fix: resolve ts-ignore bug in DIDropDownView
- Caused by handling wrong userselect css option
- Caused by empty parameter

bonomoon committed:
fix: resolve ts-ignore bug in highlightCell, DIDropDownView
- Caused by not using correct selected attribute
- Caused by not using correct function name

bonomoon committed:
chore: convert to arrow function in DeskEvent listeners, DIDropDownView

bonomoon committed:
chore: resolve @typescript-eslint/no-unsafe-type in DICanvas
- Define CanvasContextId type for initiating Canvas object

bonomoon committed:
Merge remote-tracking branch 'origin/develop' into refactoring/resolve-all-eslint-issues-and-remove-inline-disables

# Conflicts:
#	System/Desk/DIView.ts

bonomoon committed:
chore: resolve @typescript-eslint/no-unusual-vars comments

bonomoon committed:
chore: resolve eslint-disable-next-line/class-methods-use-this

bonomoon committed:
chore: use arrow function to bind this, not using bind function

bonomoon committed:
chore: refactoring code structure

qlqhwb commented on System/Desk/DITableCell.ts:
```diff
@@ -9,20 +9,15 @@ export class DITableCell {
 
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    constructor(className?: string, idName?: string) {
+    constructor(className?: string, _idName?: string) {
```
is there a particular reason for this change?

qlqhwb commented on System/Desk/DIListView.ts:
```diff
@@ -25,19 +25,33 @@ export class DIListView extends DIView {
+            this.delegate = delegate;
+        }
+
+        switch ((this.cellClickType = cellClickType)) {
```
is there any reason for using doubled parenthesis?

qlqhwb commented on System/Desk/Desk.ts:
```diff
@@ -343,41 +326,37 @@ export class Desk {
-                }
+
+                secretaryInstance.mainWorkSpace.apps.forEach((app) => app.allowDrag && app.dragEnd(false));
+
```
I think it's better to do it more explicitly

bonomoon replied:
Got it. :)

qlqhwb commented on System/Drawer/Drawer.ts:
```diff
@@ -826,7 +826,8 @@ export class Drawer extends Application {
             },
-            this.updateUploadFile.bind(this),
+            // @ts-ignore
+            (file: DeskFile, loaded: boolean, total: number) => this.updateUploadFile(file, loaded / total),
```
this should have parameters file: DeskFileUpload, and progress: number. No these three

qlqhwb commented on System/Secretary/FileManager.ts:
```diff
@@ -160,11 +151,11 @@ export class FileManager {
                     // 404 not found!
-                    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-                    const err = Object.freeze({ type: 2, message: "File does not exist" });
+                    const _err = Object.freeze({ type: 2, message: "File does not exist" });
```
is there any reason for this change?

bonomoon replied:
This is just to express unused variable.

qlqhwb commented on System/Secretary/FileManager.ts:
```diff
@@ -603,16 +594,11 @@ export class FileManager {
-        }
-        req.addData("FileList[]", fileIds);
+
+        req.addData("FileList[]", this.createFileIdList(files));
```
I thinks this can just be inline instead of having separate function.
Also, we know the type of the file. So, it should be changed in there and the function parameters.

qlqhwb replied:
I see it's being used multiple places, so maybe just edit the type.

qlqhwb commented on System/Secretary/RequestServer.ts:
```diff
@@ -93,8 +100,7 @@ export class RequestServer {
 
-    // eslint-disable-next-line @typescript-eslint/no-unused-vars
-    callListeners(response: any, err: any) {
+    callListeners(response: any, _err: any) {
```
any reason for this change?

bonomoon replied:
Parameter with underscore means unused variable in javascript.

qlqhwb commented on System/Secretary/Secretary.ts:
```diff
@@ -326,51 +327,39 @@ export class Secretary {
     }
 
-    urlForFileId(fileId: any) {
+    urlForFileId(fileId: string) {
```
this is being used as a string for now. But it was meant to be any type. So if casting is required, it should be done inside of the function.

mattddean replied:
Let's try an `unknown` type for fileId

bonomoon replied:
@mattddean I understand your intention, but `unknown` type will make error if this method is invoked with parameters which are not `any` or `unknown` type on typescipt system.

bonomoon replied:
@qlqhwb I think it seems to be used as a `string` or `number` type, anything else?

mattddean replied:
https://www.typescriptlang.org/docs/handbook/declaration-files/do-s-and-don-ts.html#any

We fall in the "want to accept anything because you will be blindly passing it through without interacting with it" category because we want to blindly pass fileId through to a backend that we didn't write, which may want to have fileId be anything, even an object.

qlqhwb commented on System/Secretary/Secretary.ts:
```diff
@@ -326,51 +327,39 @@ export class Secretary {
     }
 
-    loadFileWithId(fileId: any, onCompletion: (data: Blob, error: any) => void, sync?: boolean) {
+    loadFileWithId(fileId: number, onCompletion: (data: Blob, error?: any) => void, sync?: boolean) {
```
fileId is any

mattddean replied:
Let's try an `unknown` type for fileId

bonomoon replied:
@qlqhwb It is the same thing like above.
